src/Tribe/Admin_List.php

```diff
@@ -155,6 +158,41 @@ public static function sort_by_event_date( Array $clauses, WP_Query $wp_query )
 			return $clauses;
 		}
 
+		/**
+		 * Defines custom logic for filtering events table by aggregator record.
+		 *
+		 * @param array    $clauses    SQL clauses for fetching posts.
+		 * @param WP_Query $wp_query   A paginated query for items.
+		 *
+		 * @return array                  Modified SQL clauses.
+		 */
+		public static function filter_by_aggregator_record( array $clauses, WP_Query $wp_query ) {
+			// Check for event post type.
+			if ( $wp_query->get( 'post_type' ) !== Tribe__Events__Main::POSTTYPE ) {
+				return $clauses;
+			}
+
+			// Check if filtering by aggregator record.
+			$record_num = intval( tribe_get_request_var( 'aggregator_record', 0 ) );
+			if ( 0 === $record_num ) {
+				return $clauses;
+			}
+
+			global $wpdb;
+
+			// Add the record meta query if it is missing.
+			if ( ! preg_match( '/tribe_aggregator_record/', $clauses['join'] ) ) {
+				$clauses['join'] .= " LEFT JOIN {$wpdb->postmeta} AS tribe_aggregator_record ON {$wpdb->posts}.ID = tribe_aggregator_record.post_id AND tribe_aggregator_record.meta_key = '_tribe_aggregator_parent_record' ";
```

Comment:
The table alias, here `tribe_aggregator_record` should either by dynamic (i.e. a base slug and a number) or be less. generic. We've had issues in the past where using too generic table aliases in queries would come back and haunt use.

E.g.

```suggestion
$table_alias = 'ea_record_' . substr( uniqid( 'ea_record', true ), 0, 10 );

				$clauses['join'] .= " LEFT JOIN {$wpdb->postmeta} AS {$table_alias} ON {$wpdb->posts}.ID = tribe_aggregator_record.post_id AND tribe_aggregator_record.meta_key = '_tribe_aggregator_parent_record' ";
```

In the `where` clause, do the same and use a dynamic table name.

If you want to keep using a fixed string table alias, then make it less generic and more specific to the operation.